src/PoolManager.sol

```diff
@@ -320,24 +320,37 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
     }
 
     /// @inheritdoc IPoolManager
-    function settle(Currency currency) external payable override noDelegateCall onlyByLocker returns (uint256 paid) {
-        uint256 reservesBefore = reservesOf[currency];
-        reservesOf[currency] = currency.balanceOfSelf();
-        paid = reservesOf[currency] - reservesBefore;
-        // subtraction must be safe
-        _accountDelta(currency, -(paid.toInt128()));
+    function mint(Currency currency, address to, uint256 amount) external override noDelegateCall onlyByLocker {
+        _accountDelta(currency, amount.toInt128());
+
+        /// Not using internal _mint to avoid totalSupply update
+        uint256 id = currency.toId();
+        balanceOf[to][id] += amount;
+        emit Transfer(msg.sender, address(0), to, id, amount);
     }
 
     /// @inheritdoc IPoolManager
-    function mint(Currency currency, address to, uint256 amount) external noDelegateCall onlyByLocker {
-        _accountDelta(currency, amount.toInt128());
-        _mint(to, currency, amount);
+    function burn(Currency currency, address from, uint256 amount) external override noDelegateCall onlyByLocker {
+        _accountDelta(currency, -(amount.toInt128()));
+
+        uint256 id = currency.toId();
+        if (from != msg.sender && !isOperator[from][msg.sender]) {
+            uint256 senderAllowance = allowance[from][msg.sender][id];
+            if (senderAllowance != type(uint256).max) {
+                allowance[from][msg.sender][id] = senderAllowance - amount;
+            }
+        }
+        balanceOf[from][id] -= amount;
+        emit Transfer(msg.sender, from, address(0), id, amount);
     }
 
     /// @inheritdoc IPoolManager
-    function burn(Currency currency, uint256 amount) external noDelegateCall onlyByLocker {
-        _accountDelta(currency, -(amount.toInt128()));
-        _burn(currency, amount);
+    function settle(Currency currency) external payable override noDelegateCall onlyByLocker returns (uint256 paid) {
```

Comment:
can you put `settle` back up before `mint` and `burn` please? Unless theres a reason to change the order? The diff is harder to review

Reply:
Sure!